src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Progress/CodeAnalysisProgressExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis;
+
+internal static class CodeAnalysisProgressExtensions
```

Comment:
these are a bunch of extensions i added to make this PR/diff much simpler.  It maps the old IProgressTRacker api onto `IProgress<>`, allowing most code to not have to change

Reply:
followup pr will remove.